runtime/cudaq/algorithms/evolve.h

```diff
@@ -64,6 +65,8 @@ convertOps(const std::initializer_list<OpTy> &ops) {
 template <typename HamTy,
           typename CollapseOpTy = cudaq::operator_sum<cudaq::matrix_operator>,
           typename ObserveOpTy = cudaq::operator_sum<cudaq::matrix_operator>>
+  requires cudaq::operator_type<HamTy> && cudaq::operator_type<CollapseOpTy> &&
```

Comment:
since we have to work with C++17 and C++20, this will need to have an equivalent SFINAE version instead of constraints (which only work for C++20)